File: easiness-core/src/main/java/com/easiness/core/entities/PeopleEntity.java
package com.easiness.core.entities;

import com.easiness.core.common.enums.PeopleType;
import lombok.*;

import javax.persistence.*;
import java.math.BigDecimal;
import java.util.List;

@Entity
@Getter
@Setter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "people", uniqueConstraints = {@UniqueConstraint(columnNames = {"name", "company_name"})})
public class PeopleEntity extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    @Column(name = "id", nullable = false, updatable = false)
    Long id;

    @Column(name = "name")
    String name;

    @Column(name = "company_name")
    String companyName;

    @Column(name = "address")
    String address;

    @Column(name = "email")
    String email;

    @Enumerated(EnumType.STRING)
    @Column(name = "type")
    PeopleType type;

    @Column(name = "balance")
    BigDecimal balance;

    @OneToMany(fetch = FetchType.EAGER, cascade = { CascadeType.ALL }, orphanRemoval = true)
    @JoinColumn(name = "owner_id", referencedColumnName = "id")
    List<ContactNoEntity> contactNoList;
}

File: easiness-ui/app/server/entity/people.entity.ts
import * as Big from "big.js"
import { Entity, PrimaryGeneratedColumn, Column, Unique, OneToMany } from "typeorm"
import { PeopleType } from "../model/people.model"
import { BigColumn } from "../utils/decorators"
import { Base } from "./base.entity"
import { ContactNo } from "./contact-no.entity"


export const uniquePeopleCols = ['name', 'companyName']

@Entity({ name: 'people' })
@Unique(uniquePeopleCols)
export class People extends Base {
    @PrimaryGeneratedColumn()
    id: number

    @Column()
    name: string

    @Column()
    companyName: string

    @Column({ nullable: true })
    address!: string

    @Column()
    type: PeopleType

    @Column({ nullable: true })
    email!: string

    @BigColumn()
    balance: Big

    @OneToMany(() => ContactNo, (contact) => contact.owner, { eager: true, cascade: true })
    contactNoList: ContactNo[]
}

